Automattic/wp-calypso
# Checkout: Use string interpolation for tax names in VAT form

## Proposed Changes

The checkout billing info/contact info form sometimes collects VAT information depending on the country chosen. Because the name of VAT changes based on country, it would be better to use a localized term. In https://github.com/Automattic/wp-calypso/pull/73686 we modified the labels for the VAT form use use the more generic term "Business Tax", but this is still a little unclear.

In this PR we switch each of these labels to replace "Business Tax" with the selected country's term for VAT (eg: "GST", "CT", etc.).

**NOTE:** this adds a good number of new strings. Make sure to translate these before merging!

**NOTE 2:** this does not alter the labels of other pages that interact with VAT, like `/me/purchases/vat-details`, because on those pages it's not as easy to tell what country we should use to determine the label. Updating those (if possible) will be the subject of a different PR.

Before:

<img width="570" alt="Screenshot 2023-03-16 at 4 05 16 PM" src="https://user-images.githubusercontent.com/2036909/225740277-00db52ee-85f1-437c-a617-69450d1af7b1.png">

After:

<img width="578" alt="Screenshot 2023-03-16 at 4 04 27 PM" src="https://user-images.githubusercontent.com/2036909/225740262-9bccf458-de14-4fcf-ae6c-60335851af84.png">

<img width="571" alt="Screenshot 2023-03-16 at 4 07 49 PM" src="https://user-images.githubusercontent.com/2036909/225740859-d640ff36-1c80-4d98-8df2-8a4e7ff6e535.png">


## Testing Instructions

- Add a product to your cart and visit checkout.
- Click to edit the billing details step if it is not already active.
- Select "United Kingdom" as the country.
- Check the VAT checkbox.
- Verify that the form mentions "VAT".
- Change the country to "Switzerland".
- Check the GST checkbox.
- Verify that the form mentions "GST".
